patches/tModLoader/Terraria/ModLoader/GlobalNPC.cs

```diff
@@ -33,6 +33,16 @@ protected sealed override void Register()
 
 	public sealed override void SetupContent() => SetStaticDefaults();
 
+	/// <summary>
+	/// Similar to SetDefaults, but only called on NPCs with a negative <see cref="NPC.netID"/>.<br/>
+	/// Vanilla "variant" NPCs which have <see cref="NPC.netID"/> set to a negative value cannot be properly modified through SetDefaults.<br/>
+	/// This method allows you to properly run code you would run in SetDefaults for these NPCs.<br/>
+	/// <b><see cref="NPC.type"/> does not support negative numbers! Always be sure to check against <see cref="NPC.netID"/> instead.</b><br/>
+	/// </summary>
```

Comment:
```suggestion
	/// <summary>
	/// Called after SetDefaults for NPCs with a negative <see cref="NPC.netID"/><br/>
	/// This hook is required because <see cref="NPC.SetDefaultsFromNetId"/> only sets <see cref="NPC.netID"/> after SetDefaults<br/>
	/// Remember that <see cref="NPC.type"/> does not support negative numbers and AppliesToEntity cannot distinguish between NPCs with the same type but differet netID<br/>
	/// </summary>
```